### PR TITLE
docker container: dockerfile update 

### PR DESCRIPTION
**Pull request by shiqizng**
updating dockerfile to support devmode + followermode container
- supporting private network topology override.  
- adding a new env var `PEER_ADDRESS` for running multiple containers. A follower node container can connect to the relay node container by setting the host:port in`PEER_ADDRESS` .

### CODE REVIEW

File: docker/files/run/run.sh [hunk 1]
```diff
@@ -59,7 +64,12 @@ function configure_data_dir() {
 
   # check for token overrides
   if [ "$TOKEN" != "" ]; then
+    # set token for relay node
     echo "$TOKEN" >algod.token
+    # set token for follower node
+    if [ -d "${ALGORAND_DATA}/../follower/" ]; then
+       echo "$TOKEN" >"${ALGORAND_DATA}/../follower/algod.token"
+    fi
```
Comment (winder) [inline docker/files/run/run.sh:72]: Is there a more general way to do this? What if the template has 3 nodes and different names?

File: docker/README.md [hunk 2]
```diff
@@ -42,8 +43,7 @@ Configuration can be modified by specifying certain files. These can be changed
 | /etc/algorand/algod.token | Override default randomized REST API token. |
 | /etc/algorand/algod.admin.token | Override default randomized REST API admin token. |
 | /etc/algorand/logging.config | Use a custom [logging.config](https://developer.algorand.org/docs/run-a-node/reference/telemetry-config/#configuration) file for configuring telemetry. |
-
-TODO: `/etc/algorand/template.json` for overriding the private network topology.
+ | /etc/algorand/template.json | Override default private network topology. |
```
Comment (winder) [inline docker/README.md:46]:
```suggestion
 | /etc/algorand/template.json | Override default private network topology. One of the nodes in the template must be named "data". | 
```

Comment (shiqizng) [reply docker/README.md:46]: that's right.

File: docker/files/run/run.sh (hunk 1 above)
Comment (shiqizng) [reply docker/files/run/run.sh]: updated.

File: docker/README.md (hunk 2 above)
Comment (shiqizng) [reply docker/README.md]: updated

**Review by winder — APPROVED** [review]
lgtm

**Review by excalq — APPROVED** [review]

File: docker/README.md [hunk 3]
```diff
@@ -20,17 +20,18 @@ By default the following config.json overrides are applied:
...
+| FAST_CATCHUP  | If set to 1 on a public network, attempt to start fast-catchup during initial config.                                                               |
+| TELEMETRY_NAME| If set on a public network, telemetry is reported with this name.                                                                                   |
+| DEV_MODE      | If set to 1 on a private network, enable dev mode. Only used during data directory initialization.                                                  |
+| NUM_ROUNDS    | If set on a private network, override default of 30000 participation keys.                                                                          |
+| TOKEN         | If set, overrides the REST API token.                                                                                                               |
+| ADMIN_TOKEN   | If set, overrides the REST API admin token.                                                                                                         |
+| KMD_TOKEN | If set along with `START_KMD`, override the KMD REST API token.                                                                                     |
```
[the remark this answers is not in the document]
Comment (shiqizng) [reply docker/README.md]: fixed

**Review by winder — APPROVED** [review]

**Review by excalq — APPROVED** [review]

**Pull request merged**